test/common/http/utility_test.cc

```diff
@@ -357,6 +357,26 @@ initial_connection_window_size: 65535
   }
 }
 
+TEST(HttpUtility, TestDefaultsProfile) {
+  std::unique_ptr<ScopedDefaultsProfileSingleton> dp =
+      std::make_unique<ScopedDefaultsProfileSingleton>(std::make_unique<DefaultsProfile>());
+  DefaultsProfile::ConfigContext context =
+      DefaultsProfile::ConfigContext("envoy.extensions.filters.network.http_connection_manager.v3."
+                                     "HttpConnectionManager.http2_protocol_options");
+
+  envoy::config::core::v3::Http2ProtocolOptions http2_options;
+  http2_options = ::Envoy::Http2::Utility::initializeAndValidateOptions(context, http2_options);
+
+  // TODO(saeedali): hardcode correct values instead of pulling them from profile
+  EXPECT_EQ(DefaultsProfileSingleton::get().getNumber(context, "max_concurrent_streams", -1),
+            http2_options.max_concurrent_streams().value());
+  EXPECT_EQ(DefaultsProfileSingleton::get().getNumber(context, "initial_stream_window_size", -1),
```

Comment:
Can we modify `getNumber()` to specify the type of what is being fetched?
Maybe use a templated function that validates that the expected value is the correct type?
The rationale is that strong typing can save us from accidentally passing wrong values (e.g., a negative number that is being converted into unsigned).